### PR TITLE
Create docs for building your own HookOS

### PR DESCRIPTION
**Pull request by chrisnegus**
*Description of changes:*

The procedure in this PR describes how to create a custom HookOS to use with EKS Anywhere Bare Metal. It also removes the file which was scheduled to hold the Custom OS procedure. But that procedure will be added at a later date.

### CODE REVIEW

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md [hunk 1]
```diff
@@ -3,6 +3,168 @@ title: "Customize HookOS for EKS Anywhere on Bare Metal"
...
+    +  cd tink-docker; docker buildx build —platform linux/amd64 —push -t $(ORG)/hook-docker:0.0 .
+    ```
+
+    Changes above change the ORG variable to use a local registry (`localhost:5000`) and change the `docker build` command to only build for the immediately required platform to save time.
+
+
+1. Modify the `hook.yaml` file located in the root of the repo with the following changes:
+
+    ```bash
+    *diff --git a/hook.yaml b/hook.yaml*
```
Comment (ptrivedi) [inline docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:66]: Hmm, what are these asterisks here and down below. They are showing up like that on the rendered page as well, might be a copy/paste thing from quip

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md [hunk 2]
```diff
@@ -3,6 +3,168 @@ title: "Customize HookOS for EKS Anywhere on Bare Metal"
...
+1. Ensure that the `linuxkit` tool is in your PATH:
+
+    ```bash
+    export PATH=$PATH:/home/tink/linuxkit/bin
+    ```
+
+1. Start a local registry:
+
+    ```bash
+    docker run -d -p 5000:5000 -—name *registry* *registry*:2
```
Comment (ptrivedi) [inline docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:153]: Need to get rid of the asterisks around the word 'registry' above as well

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md [hunk 3]
```diff
@@ -3,6 +3,168 @@ title: "Customize HookOS for EKS Anywhere on Bare Metal"
...
+1. Artifacts will be put under the `dist` directory in the repo’s root:
+
+    ```bash
+    ./initramfs-aarch64
+    ./initramfs-x86_64
+    ./vmlinuz-aarch64
+    ./vmlinuz-x86_64
+    ```
+
+1. To use the kernel (`vmlinuz`) and initial ram disk (`initramfs`) when you build your cluster, see the description of the `hookImagesURLPath` field in your [Bare Metal configuration]({{< relref "../clusterspec/baremetal/" >}}) file.
```
Comment (ptrivedi) [inline docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:170]: We should point this link to the exact anchor for the hookImagesURLPath field description

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md (hunk 1 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:66]: Yes. I think they are bold markings. I missed that. I'll remove them.

Comment (chrisnegus) [reply docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:66]: Done.

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md (hunk 2 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:153]: Done.

File: docs/content/en/docs/reference/baremetal/bare-custom-hookos.md (hunk 3 above)
Comment (chrisnegus) [reply docs/content/en/docs/reference/baremetal/bare-custom-hookos.md:170]: Done.

**Pull request merged**